hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -434,7 +434,11 @@ boolean needsBalance(TableName tableName, BalancerClusterState cluster) {
       return true;
     }
 
-    if (sloppyRegionServerExist(cs)) {
+    if (
+      !balancerConditionals.isTableIsolationEnabled() // table isolation is inherently incompatible
+                                                      // with naive "sloppy server" checks
+        && sloppyRegionServerExist(cs)
+    ) {
```

Comment:
In reality, I still think we will fix "sloppy" servers more effectively than a non-conditional balancer implementation by way of the SlopFixingCandidateGenerator (which _is_ table isolation compatible, and performs better than the cost function based approach in my testing)